### PR TITLE
Add sensors section to describe different modes of sensors

### LINKED ISSUE #10816: Docs around different execution modes for Sensor
**Description**

It would be good to add docs to explain different modes for Sensors:

1. Poke mode
1. Reschedule mode
1. Smart Sensor

and to explain the advantages of one over the other

### PR DESCRIPTION
**Pull request by xinbinhuang**
closes: #10816 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by xinbinhuang** [comment]
cc: @ryw  @kaxil

File: docs/apache-airflow/concepts.rst [hunk 1]
```diff
@@ -448,7 +447,81 @@ Operators are only loaded by Airflow if they are assigned to a DAG.
...
+|                    | running process for   | infrastructure cost   |
+|                    | each sensor and       | and improve cluster   |
+|                    | poking periodically,  | stability - reduce    |
+|                    | a sensor will only    | meta database load.   |
+|                    | store poke context at |                       |
+|                    | ``sensor_instance``   |                       |
+|                    | table and then exits  |                       |
+|                    | with a 'sensing'      |                       |
+|                    | state.                |                       |
++--------------------+-----------------------+-----------------------+
```
Comment (xinbinhuang) [inline docs/apache-airflow/concepts.rst:504]: For this table, it's shown with a scrollbar horizontally and it's a bit long. Is there a way to wrap the text inside the cell instead?

Comment (ashb) [reply docs/apache-airflow/concepts.rst:504]: I'd probably suggest not making it a table then

**Comment by xinbinhuang** [comment]
Also cc Airbnb: @YingboWang @KevinYang21

File: docs/apache-airflow/concepts.rst (hunk 1 above)
Comment (xinbinhuang) [reply docs/apache-airflow/concepts.rst:504]: hmm... if that's the case, I will remove the table later

**Review by kaxil — APPROVED** [review]
I think this is a good start. We can iterate as we go

**Pull request merged**